freqtrade/freqai/data_drawer.py

```diff
@@ -163,37 +164,67 @@ def load_historic_predictions_from_disk(self):
         Locate and load a previously saved historic predictions.
         :return: bool - whether or not the drawer was located
         """
-        exists = self.historic_predictions_path.is_file()
+        exists = self.historic_predictions_folder.exists()
+        convert = self.historic_predictions_path.is_file()
+
         if exists:
             try:
-                with self.historic_predictions_path.open("rb") as fp:
-                    self.historic_predictions = cloudpickle.load(fp)
+                self.load_historic_predictions_from_folder()
                 logger.info(
                     f"Found existing historic predictions at {self.full_path}, but beware "
                     "that statistics may be inaccurate if the bot has been offline for "
                     "an extended period of time."
                 )
             except EOFError:
                 logger.warning(
-                    'Historical prediction file was corrupted. Trying to load backup file.')
-                with self.historic_predictions_bkp_path.open("rb") as fp:
-                    self.historic_predictions = cloudpickle.load(fp)
-                logger.warning('FreqAI successfully loaded the backup historical predictions file.')
+                    'Historical prediction files were corrupted. Trying to load backup files.')
+                self.load_historic_predictions_from_folder()
+                logger.warning('FreqAI successfully loaded the backup '
+                               'historical predictions files.')
+
+        elif not exists and convert:
+            logger.info("Converting your historic predictions pkl to parquet"
+                        "to improve performance.")
+            with Path.open(self.historic_predictions_path, "rb") as fp:
+                self.historic_predictions = cloudpickle.load(fp)
+            self.save_historic_predictions_to_disk()
+            exists = True
 
         else:
-            logger.info("Could not find existing historic_predictions, starting from scratch")
+            logger.warning(
+                f"Follower could not find historic predictions at {self.full_path} "
+                "sending null values back to strategy"
+            )
 
         return exists
 
+    def load_historic_predictions_from_folder(self):
+        """
+        Try to build the historic_predictions dictionary from parquet
+        files in the historic_predictions_folder
+        """
+        for file_path in self.historic_predictions_folder.glob("*.parquet"):
+            key = file_path.stem
+            key.replace("_", "/")
```

Comment:
same as below - but in reverse (`idatahandler.rebuild_pair_from_filename()` has logic for this ... but it's not readily usable iirc due to it's location, so would need either copy/paste or a small refactoring to extract it from that class, potentially into misc alongside the pair_to_filename method)